lib/linebuffer.js

```diff
@@ -34,12 +34,13 @@ const needsRetry = (error, response, body) => {
 
     if (!response) { return false; }
 
-    if (response.statusCode !== 200) {
-        debug(`req failed: ${body}`);
-        if (bufferStats.flushcount % 10 === 0) { utils.log(`request failed: ${body}`); }
+    if (!response.statusCode >= 400) {
+        debug(`[${response.statusCode}] req failed: ${JSON.stringify(body)}`);
+        if (bufferStats.flushcount % 10 === 0) { utils.log(`[${response.statusCode}] request failed: ${JSON.stringify(body)}`); }
```

Comment:
Should just call `stringify` once and save as local variable:
```
const bodyString = JSON.stringify(body);
```

Reply:
@weizou19, nice suggestion! updated!